extensions/ocs/pkg/service/v0/users.go

```diff
@@ -97,6 +108,95 @@ func (o Ocs) GetUser(w http.ResponseWriter, r *http.Request) {
 	o.mustRender(w, r, response.DataRender(d))
 }
 
+func (o Ocs) fillPersonalQuota(ctx context.Context, d *data.User, u *cs3identity.User) {
+
+	gc, err := pool.GetGatewayServiceClient(o.config.Reva.Address)
+	if err != nil {
+		o.logger.Error().Err(err).Msg("error getting gateway client")
+		return
+	}
+
+	aRes, err := gc.Authenticate(ctx, &cs3gateway.AuthenticateRequest{
+		Type:         "machine",
+		ClientId:     "userid:" + u.Id.OpaqueId,
+		ClientSecret: o.config.MachineAuthAPIKey,
+	})
+
+	switch {
+	case err != nil:
+		o.logger.Error().Err(err).Msg("could not fill personal quota")
+		return
+	case aRes.Status.Code != cs3rpc.Code_CODE_OK:
+		o.logger.Error().Interface("status", aRes.Status).Msg("could not fill personal quota")
+	}
+
+	ctx = metadata.AppendToOutgoingContext(ctx, revactx.TokenHeader, aRes.Token)
+
+	res, err := gc.ListStorageSpaces(ctx, &cs3storage.ListStorageSpacesRequest{
+		Filters: []*cs3storage.ListStorageSpacesRequest_Filter{
+			{
+				Type: cs3storage.ListStorageSpacesRequest_Filter_TYPE_OWNER,
+				Term: &cs3storage.ListStorageSpacesRequest_Filter_Owner{
+					Owner: aRes.User.Id,
+				},
+			},
+			{
+				Type: cs3storage.ListStorageSpacesRequest_Filter_TYPE_SPACE_TYPE,
+				Term: &cs3storage.ListStorageSpacesRequest_Filter_SpaceType{
+					SpaceType: "personal",
+				},
+			},
+		},
+	})
+	if err != nil {
+		o.logger.Error().Err(err).Msg("error calling ListStorageSpaces")
+		return
+	}
+	if res.Status.Code != cs3rpc.Code_CODE_OK {
+		o.logger.Debug().Interface("status", res.Status).Msg("ListStorageSpaces returned non OK result")
+		return
+	}
+
+	if len(res.StorageSpaces) == 0 {
+		o.logger.Debug().Err(err).Msg("list spaces returned empty list")
+		return
+	}
+
+	getQuotaRes, err := gc.GetQuota(ctx, &cs3gateway.GetQuotaRequest{Ref: &cs3storage.Reference{
+		ResourceId: res.StorageSpaces[0].Root,
+		Path:       ".",
+	}})
+	if err != nil {
+		o.logger.Error().Err(err).Msg("error calling GetQuota")
+		return
+	}
+	if res.Status.Code != cs3rpc.Code_CODE_OK {
+		o.logger.Debug().Interface("status", res.Status).Msg("GetQuota returned non OK result")
+		return
+	}
+
+	total := getQuotaRes.TotalBytes
+	used := getQuotaRes.UsedBytes
+
+	d.Quota = &data.Quota{
+		Used: int64(used),
+		// TODO support negative values or flags for the quota to carry special meaning: -1 = uncalculated, -2 = unknown, -3 = unlimited
+		// for now we can only report total and used
+		Total: int64(total),
```

Comment:
the oc10 code has this: https://github.com/owncloud/core/blob/817f54f53d64249166feaa0eb9231c50ca1ed7e0/lib/public/Files/FileInfo.php#L44-L58

it is returned for `quota-available-bytes` in propfind requests: 
```xml
 $ curl https://demo.owncloud.com/remote.php/webdav/ -udemo:demo -X PROPFIND -d '<?xml version="1.0" ?><d:propfind xmlns:d="DAV:"><d:prop><d:quota-available-bytes/><d:quota-used-bytes/></d:prop></d:propfind>' -v --trace-time | xmllint -format -

...

  <d:response>
    <d:href>/remote.php/webdav/Photos/</d:href>
    <d:propstat>
      <d:prop>
        <d:quota-available-bytes>-3</d:quota-available-bytes>
        <d:quota-used-bytes>1011464</d:quota-used-bytes>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
...
```

the ocs api only returns quota on the `/ocs/v[1|2].php/cloud/users/{username}` endpoint:
```json
$ curl https://demo.owncloud.com/ocs/v1.php/cloud/users/demo?format=json -u demo:demo -v | jq

...

      "quota": {
        "free": 27273637888,
        "used": 7716359,
        "total": 27281354247,
        "relative": 0.03,
        "definition": "none"
      },

...

```

but it always returns the actually available bytes, not -3 ...